src/plots/smith/layout_attributes.js

```diff
@@ -0,0 +1,235 @@
+    },
+
+
+    title: {
+        // radial title is not gui-editable at the moment,
+        // so it needs dflt: '', similar to carpet axes.
+        text: extendFlat({}, axesAttrs.title.text, {editType: 'plot', dflt: ''}),
+        font: extendFlat({}, axesAttrs.title.font, {editType: 'plot'}),
+
+        // TODO
+        // - might need a 'titleside' and even 'titledirection' down the road
+        // - what about standoff ??
+
+        editType: 'plot'
+    },
+
+    hoverformat: axesAttrs.hoverformat,
+
+    uirevision: {
+        valType: 'any',
+        editType: 'none',
+        description: [
+            'Controls persistence of user-driven changes in axis `range`,',
+            '`autorange`, `angle`, and `title` if in `editable: true` configuration.',
+            'Defaults to `polar<N>.uirevision`.'
+        ].join(' ')
+    },
+
+    editType: 'calc',
+
+    _deprecated: {
+        title: axesAttrs._deprecated.title,
+        titlefont: axesAttrs._deprecated.titlefont
+    }
+};
+
+extendFlat(
+    radialAxisAttrs,
+
+    // N.B. realaxis grid lines are circular,
+    // but realaxis lines are straight from circle center to outer bound
+    axisLineGridAttr,
+    axisTickAttrs
+);
+
+var angularAxisAttrs = {
+    visible: extendFlat({}, axesAttrs.visible, {dflt: true}),
+
+    direction: {
+        valType: 'enumerated',
+        values: ['counterclockwise', 'clockwise'],
+        dflt: 'counterclockwise',
+        editType: 'calc',
+        description: [
+            'Sets the direction corresponding to positive angles.'
+        ].join(' ')
+    },
+
+    rotation: {
+        valType: 'angle',
+        editType: 'calc',
+        description: [
+            'Sets that start position (in degrees) of the angular axis',
+            'By default, polar subplots with `direction` set to *counterclockwise*',
+            'get a `rotation` of *0*',
+            'which corresponds to due East (like what mathematicians prefer).',
+            'In turn, polar with `direction` set to *clockwise* get a rotation of *90*',
+            'which corresponds to due North (like on a compass),'
+        ].join(' ')
+    },
+
+    hoverformat: axesAttrs.hoverformat,
+
+    uirevision: {
+        valType: 'any',
+        editType: 'none',
+        description: [
+            'Controls persistence of user-driven changes in axis `rotation`.',
+            'Defaults to `polar<N>.uirevision`.'
+        ].join(' ')
+    },
+
+    editType: 'calc'
+};
+
+extendFlat(
+    angularAxisAttrs,
+
+    // N.B. angular grid lines are straight lines from circle center to outer bound
+    // the angular line is circular bounding the polar plot area.
+    axisLineGridAttr,
+
+    // N.B. ticksuffix defaults to '°' for angular axes with `thetaunit: 'degrees'`
+    axisTickAttrs
+);
+
+module.exports = {
+    // TODO for x/y/zoom system for paper-based zooming:
+    // x: {},
+    // y: {},
+    // zoom: {},
+
+    domain: domainAttrs({name: 'polar', editType: 'plot'}),
```

Comment:
We may need to change `polar` to `smith` here and in several places above and below.

Reply:
See eddfcde022b667bf69b123ad4258bb1ebfba53d3.

Reply:
> See [eddfcde](https://github.com/plotly/plotly.js/commit/eddfcde022b667bf69b123ad4258bb1ebfba53d3).

This one did not addressed there.

Reply:
5605f78ee8a3dd55c61283f2c93f60cff257d59b